* Parametrize collision checking in nav2_graceful_controller

suchetanrs opened the pull request:
## Basic Info
| ------ | ----------- |
| Ticket(s) this addresses   | (add tickets here #5005 ) |
| Primary OS tested on | (Ubuntu 24.04 running Jazzy) |
| Robotic platform tested on | (Simulation of a custom diff-drive robot) |
| Does this PR contain AI generated software? | (No) |

---

## Description of contribution in a few bullet points

* I added a parameter to nav2_graceful_controller to enable/disable collision checking

## Description of documentation updates required from your changes

* Added new parameter, so need to add that to default configs and documentation page.

## Description of how this change was tested

* I ran the test with a huge robot radius in the local_costmap to always have collision in the controller's path. The parametrization worked to enable/disable it.

#### For Maintainers: <!-- DO NOT EDIT OR REMOVE -->
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

SteveMacenski commented on nav2_graceful_controller/src/graceful_controller.cpp:
```diff
@@ -177,7 +180,7 @@ geometry_msgs::msg::TwistStamped GracefulController::computeVelocityCommands(
       geometry_msgs::msg::PoseStamped costmap_pose;
       tf2::doTransform(next_pose, costmap_pose, costmap_transform);
-      if (inCollision(
+      if (use_collision_detection && inCollision(
```
```suggestion
      if (params_->use_collision_detection && inCollision(
```

And remove L174 where `use_collision_detection` is set, that's not needed

SteveMacenski reviewed (commented):
LGTM, please update the docs https://docs.nav2.org/configuration/packages/configuring-graceful-motion-controller.html to include this new parameter and happy to merge the pair with a small adjustment

suchetanrs commented:
Thanks a lot for your response!
I have addressed your comment and also added the parameter to dynamic reconfigure.

A PR for the docs is raised here: https://github.com/ros-navigation/docs.nav2.org/pull/660

SteveMacenski commented:
@suchetanrs thanks and merged!
